pqarrow/arrow.go

```diff
@@ -59,26 +59,36 @@ func ParquetSchemaToArrowSchema(ctx context.Context, schema *parquet.Schema, s *
 		}
 	}
 
-	for _, distinctExpr := range options.DistinctColumns {
-		if distinctExpr.Computed() {
-			// Usually we would pass the logical query plan as the data type
-			// finder, but we're here because of an intended layering
-			// violation, which is pushing distinct queries down to the scan
-			// layer. In this case there are no other possible physical types
-			// other than the actual schema, so we can just implement a
-			// simplified version of the type finder that doesn't need to
-			// traverse the logical plan, since this is already the physical
-			// scan layer execution.
-			dataType, err := distinctExpr.DataType(&exprTypeFinder{s: s})
-			if err != nil {
-				return nil, err
+	if len(options.DistinctColumns) > 0 {
+		for _, distinctExpr := range options.DistinctColumns {
+			if distinctExpr.Computed() {
+				// Usually we would pass the logical query plan as the data type
+				// finder, but we're here because of an intended layering
+				// violation, which is pushing distinct queries down to the scan
+				// layer. In this case there are no other possible physical types
+				// other than the actual schema, so we can just implement a
+				// simplified version of the type finder that doesn't need to
+				// traverse the logical plan, since this is already the physical
+				// scan layer execution.
+				dataType, err := distinctExpr.DataType(&exprTypeFinder{s: s})
+				if err != nil {
+					return nil, err
+				}
+				fields = append(fields, arrow.Field{
+					Name:     distinctExpr.Name(),
+					Type:     dataType,
+					Nullable: true, // TODO: This should be determined by the expression and underlying column(s).
+				})
 			}
-			fields = append(fields, arrow.Field{
-				Name:     distinctExpr.Name(),
-				Type:     dataType,
-				Nullable: true, // TODO: This should be determined by the expression and underlying column(s).
-			})
 		}
+
+		// Need to sort as the distinct columns are just appended, but we need
```

Comment:
It's been a while since I looked at this code, but is this the right place to do it? Do `ParquetSchemaToArrowSchema` consumers expect the schema to be sorted? My hunch is that the actual issue is that the builder assumes field sorting when it shouldn't and so it should be the one to sort.

If a regression test is not too much trouble I think it would be good to add.